src/data/repositories/consts/DiseaseOutbreakConstants.ts

```diff
@@ -33,3 +39,102 @@ export const DiseaseOutbreakCodes = {
+
+export type DiseaseOutbreakCode = GetValue<typeof DiseaseOutbreakCodes>;
+
+export type KeyCode = (typeof DiseaseOutbreakCodes)[keyof typeof DiseaseOutbreakCodes];
+
+export function isStringInDiseaseOutbreakCodes(code: string): code is KeyCode {
+    return Object.values(DiseaseOutbreakCodes).includes(code as KeyCode);
+}
+
+export function isHazardType(hazardType: string): hazardType is HazardType {
+    return [
+        "Biological:Animal",
+        "Biological:Human",
+        "Chemical",
+        "Environmental",
+        "Unknown",
+    ].includes(hazardType);
+}
+
+export function getValueFromDiseaseOutbreak(
+    diseaseOutbreak: DiseaseOutbreakEventBaseAttrs
+): Record<DiseaseOutbreakCode, string> {
+    return {
+        RTSL_ZEB_TEA_EVENT_id: diseaseOutbreak.eventId.toString(),
+        RTSL_ZEB_TEA_EVENT_NAME: diseaseOutbreak.name,
+        RTSL_ZEB_TEA_HAZARD_TYPE: getHazardTypeCode(diseaseOutbreak.hazardType),
+        RTSL_ZEB_TEA_MAIN_SYNDROME: diseaseOutbreak.mainSyndromeCode,
+        RTSL_ZEB_TEA_SUSPECTED_DISEASE: diseaseOutbreak.suspectedDiseaseCode,
+        RTSL_ZEB_TEA_NOTIFICATION_SOURCE: diseaseOutbreak.notificationSourceCode,
+        RTSL_ZEB_TEA_AREAS_AFFECTED_PROVINCES: getOUTextFromList(
+            diseaseOutbreak.areasAffectedProvinceIds
+        ),
+        RTSL_ZEB_TEA_AREAS_AFFECTED_DISTRICTS: getOUTextFromList(
+            diseaseOutbreak.areasAffectedDistrictIds
+        ),
+        RTSL_ZEB_TEA_INCIDENT_STATUS: diseaseOutbreak.incidentStatus,
+        RTSL_ZEB_TEA_DATE_EMERGED: diseaseOutbreak.emerged.date.toISOString(),
+        RTSL_ZEB_TEA_DATE_EMERGED_NARRATIVE: diseaseOutbreak.emerged.narrative,
+        RTSL_ZEB_TEA_DATE_DETECTED: diseaseOutbreak.detected.date.toISOString(),
+        RTSL_ZEB_TEA_DATE_DETECTED_NARRATIVE: diseaseOutbreak.detected.narrative,
+        RTSL_ZEB_TEA_DATE_NOTIFIED: diseaseOutbreak.notified.date.toISOString(),
+        RTSL_ZEB_TEA_DATE_NOTIFIED_NARRATIVE: diseaseOutbreak.notified.narrative,
+        RTSL_ZEB_TEA_INITIATE_INVESTIGATION:
+            diseaseOutbreak.earlyResponseActions.initiateInvestigation.toISOString(),
+        RTSL_ZEB_TEA_CONDUCT_EPIDEMIOLOGICAL_ANALYSIS:
+            diseaseOutbreak.earlyResponseActions.conductEpidemiologicalAnalysis.toISOString(),
+        RTSL_ZEB_TEA_LABORATORY_CONFIRMATION: diseaseOutbreak.earlyResponseActions
+            .laboratoryConfirmation.na
+            ? "true"
+            : "",
+        RTSL_ZEB_TEA_SPECIFY_DATE1:
+            diseaseOutbreak.earlyResponseActions.laboratoryConfirmation.date?.toISOString() ?? "",
+        RTSL_ZEB_TEA_APPROPRIATE_CASE_MANAGEMENT: diseaseOutbreak.earlyResponseActions
+            .appropriateCaseManagement.na
+            ? "true"
+            : "",
+        RTSL_ZEB_TEA_SPECIFY_DATE2:
+            diseaseOutbreak.earlyResponseActions.appropriateCaseManagement.date?.toISOString() ??
+            "",
+        RTSL_ZEB_TEA_APPROPRIATE_PUBLIC_HEALTH: diseaseOutbreak.earlyResponseActions
```

Comment:
I'd still abstract these two, the toISOString, and the "true", and they are used many times.